src/main/java/org/folio/search/integration/KafkaMessageListener.java

```diff
@@ -28,22 +35,61 @@
    * @param events list with the body as json from messaging system.
    */
   @KafkaListener(
-    id = "mod-search-listener",
+    id = "mod-search-instance-listener",
     containerFactory = "kafkaListenerContainerFactory",
-    topics = "${application.kafka.listener.events.topics}",
-    groupId = "${application.kafka.listener.events.group-id}",
-    concurrency = "${application.kafka.listener.events.concurrency}")
-  public void handleEvents(List<ResourceEventBody> events) {
-    log.info("Processing resource events from kafka [number of events: {}]", events.size());
+    topics = "#{'${application.kafka.listener.instances.topics}'.split(',')}",
+    groupId = "${application.kafka.listener.instances.group-id}",
+    concurrency = "${application.kafka.listener.instances.concurrency}")
+  public void handleInstanceEvents(List<ResourceEventBody> events) {
+    log.info("Processing instance events from kafka [number of events: {}]", events.size());
     var resources = events.stream()
       .map(KafkaMessageListener::asInstanceResource)
       .collect(toList());
 
     indexService.indexResources(resources);
   }
 
+  @KafkaListener(
+    id = "mod-search-events-listener",
+    containerFactory = "kafkaListenerContainerFactory",
+    topics = "#{'${application.kafka.listener.events.topics}'.split(',')}",
+    groupId = "${application.kafka.listener.events.group-id}",
+    concurrency = "${application.kafka.listener.events.concurrency}")
+  public void handleEvents(List<ConsumerRecord<String, ResourceEventBody>> consumerRecords) {
```

Comment:
Are we going to remove the `instance` handler and use only this one?

Reply:
yes. first one is temporal solution for indexing only instances, when bulk endpoint will be implement - in term of this story I will remove previous approach.